### PR TITLE
Add if block in `perf result` to avoid panic error

### LINKED ISSUE #5318: [mesheryctl] `mesheryctl perf result my-test --view` crashes when trying to see performance test results
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

#### Current Behavior
I'm trying to see the details of the performance test results I ran earlier. I run `mesheryctl perf result my-test --view`, then pick any of the results by typing let's say `0` to choose the first one, and the app crashes showing the error trace below.

#### Desired Behavior
I want to be able to see the results of the test I chose.

#### Screenshots/Logs
```
$ mesheryctl perf result my-test --view
Name: istio_1648736178355
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2bbc450]

goroutine 1 [running]:
github.com/layer5io/meshery/mesheryctl/internal/cli/root/perf.glob..func5(0x45abde0, 0xc0000a1d20, 0x1, 0x2, 0x0, 0x0)
        /Users/runner/work/meshery/meshery/mesheryctl/internal/cli/root/perf/result.go:137 +0x4b0
github.com/spf13/cobra.(*Command).execute(0x45abde0, 0xc0000a1ce0, 0x2, 0x2, 0x45abde0, 0xc0000a1ce0)
        /Users/runner/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:856 +0x472
github.com/spf13/cobra.(*Command).ExecuteC(0x45a7f60, 0x34bc020, 0x4793c30, 0x0)
        /Users/runner/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:974 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        /Users/runner/go/pkg/mod/github.com/spf13/cobra@v1.3.0/command.go:902
github.com/layer5io/meshery/mesheryctl/internal/cli/root.Execute()
        /Users/runner/work/meshery/meshery/mesheryctl/internal/cli/root/root.go:68 +0x56
main.main()
        /Users/runner/work/meshery/meshery/mesheryctl/cmd/mesheryctl/main.go:22 +0x25
```

#### Environment

- Host OS: Mac
- Meshery Server Version: stable-v0.6.0-rc-4
- Meshery Client Version: stable-v0.6.0-rc-5h
- Platform: Kubernetes

---

#### Contributor Resources
- [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by alphaX86**
Signed-off-by: Aadhitya A <aadhitya864@gmail.com>

**Description**

This PR fixes #5318

**Notes for Reviewers**
Needs tests to see if it really works in intact. Done with some pre checks given in below comment

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by alphaX86** [comment]
TL;DR

Here's the video on how this if-block works when test result is empty...

https://user-images.githubusercontent.com/59508546/161427342-fe7b39e3-9bd2-49f5-b7d8-d7a3294ad5b6.mp4

**Comment by alphaX86** [comment]
For case in issue #5318 it's bit different and I'm continuously trying to resolve it

File: mesheryctl/internal/cli/root/perf/result.go [hunk 1]
```diff
@@ -14,6 +14,7 @@ import (
 	"github.com/layer5io/meshery/mesheryctl/pkg/utils"
 	"github.com/layer5io/meshery/models"
 	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
```
Comment (hexxdump) [inline mesheryctl/internal/cli/root/perf/result.go:17]: we want to move away from logus completely. pls use utils.Log instead. Ref. #5298

Comment (alphaX86) [reply mesheryctl/internal/cli/root/perf/result.go]: Done 👍

**Comment by alphaX86** [comment]
Could anyone check with "None" provider and see if panic is disappeared when test profile is created, made to run and viewed???

File: mesheryctl/internal/cli/root/perf/result.go [hunk 2]
```diff
@@ -111,6 +111,11 @@ mesheryctl perf result saturday-profile --view
 
 		// get performance results in format of string arrays and resultStruct
 		data, expandedData := performanceResultsToStringArrays(results)
+		if len(expandedData) == 0 {
+			utils.Log.Info("No Test Results to display")
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/perf/result.go:115]:
```suggestion
			utils.Log.Info("No test results to display")
```

File: mesheryctl/internal/cli/root/perf/result.go [hunk 3]
```diff
@@ -221,8 +252,8 @@ func performanceResultsToStringArrays(results []models.PerformanceResult) ([][]s
 				P99:     P99,
 			},
 			StartTime:     result.TestStartTime,
-			MesheryID:     result.MesheryID,
-			LoadGenerator: result.RunnerResults.LoadGenerator,
+			MesheryID:     (*uuid.UUID)(mesheryid.Bytes()),
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/perf/result.go:255]: This adds support for multi-cluster, is that right, @alphaX86, @Revolyssup, @hexxdump, @piyushsingariya?

Comment (alphaX86) [reply mesheryctl/internal/cli/root/perf/result.go:255]: Yes, but if "None" provider is used, then a nil UUID is generated on both UserID and MesheryID which in turn causes panic error. That's why I made a way to bypass this so that instead of "" UUID, a blank UUID will be generated like 000-00000-0000-00000 (example)

But if it's not the case to be implemented then how can I bypass this? // @Revolyssup @hexxdump

Comment (leecalcote) [reply mesheryctl/internal/cli/root/perf/result.go:255]: @alphaX86, I recognize that its typically not the case that I have a genuine question, but am usually insinuating an answer with my question... in this case, though, this is an honest question from me. :) You might be doing the right thing. I just didn't know if this was in part for `mesheryctl` to supporting querying different Meshery Servers (different  Meshery contexts / deployments).

**Review by leecalcote — APPROVED** [review]

File: mesheryctl/internal/cli/root/perf/result.go [hunk 4]
```diff
@@ -206,10 +211,36 @@ func performanceResultsToStringArrays(results []models.PerformanceResult) ([][]s
...
+		mesheryid := uuid.Nil
+		url := "None"
+		loadGenerator := "None"
+
+		if result.UserID != nil {
+			userid = *result.UserID
+		}
+
+		if result.MesheryID != nil {
+			mesheryid = *result.MesheryID
```
Comment (leecalcote) [inline mesheryctl/internal/cli/root/perf/result.go:225]: Does this consideration need to be account for in every mesheryctl command?
